nupic/research/frameworks/pytorch/restore_utils.py

```diff
@@ -32,6 +32,99 @@
+                           Otherwise, pytorch may throw an error.
+    :param param_map: a dict mapping names of state within the checkpoint to new desired
+                      names; this helps when the names of the model's state don't quite
+                      match that of the checkpoints. Ex:
+                      `param_map = {"features.weight": "features.new_weight"}`
+                      where "features.weight" exists within the checkpoint and the model
+                      has the attribute `model.features.new_weight`.
+    :param state_dict_transform: this is a callable that takes the state_dict and
+                                 and the model and transforms the state_dict in some
+                                 way; useful for a custom re-mapping such as parameters
+                                 with new naming schemes or formats (e.g. backwards
+                                 compatibility). The output should be new state_dict.
+    """
+
+    assert os.path.isfile(chekpoint_path), (
+        "Double check the checkpoint exists and is a file."
+    )
+
+    # Load the state dict from the checkpoint.
+    state_dict = get_state_dict(chekpoint_path)
+
+    assert state_dict is not None, (
+        "Couldn't load the state_dict. "
+        "Maybe check it's in the right format ({'model': <state_dict>})"
+    )
+
+    # Remap param names in state_dict.
+    if param_map:
+        state_dict = remap_state_dict(state_dict, param_map)
+
+    # Check all desired params are present.
+    if strict and subset:
+
+        # Ensure subset is present in the checkpoint's state.
+        assert set(subset) <= set(state_dict.keys()), "".join([
+            "Found params in the subset which are not present in the checkpoint: ",
+            f"'{chekpoint_path}'"
+            "Params not present include:"
+            f"\n {set(subset) - set(state_dict.keys())}"
+        ])
+
+        # Ensure subset is present in the model's state.
+        model_params = model.state_dict()
+        assert set(subset) <= set(model_params.keys()), "".join([
+            "Found params in the subset which are not present in the model: ",
+            f"'{chekpoint_path}'"
+            "Params not present include:"
+            f"\n {set(subset) - set(model_params.keys())}"
+        ])
+
+        # Retrieve the subset of params.
+        new_state_dict = {
+            param_name: state_dict[param_name]
+            for param_name in subset
+        }
+        state_dict = new_state_dict
+        strict = False  # we now only care about the subset
+
+    # Resize the buffers of the model to match those in the state_dict.
+    if resize_buffers:
+        resize_model_buffers(model, state_dict)  # done in place
+
+    # Apply any custom transformation.
+    if state_dict_transform:
+        state_dict = state_dict_transform(state_dict, model)
+
+    # Load state.
+    model.load_state_dict(state_dict, strict=strict)
+
+    return model
+
+
```

Comment:
`load_multi_state` is not as complex as it was before, you could remove the ` # noqa: C901` from it

Reply:
Good point.